### PR TITLE
[non-record] byte jepa-lw naive experimental baseline 2.125 bpb @ 2000 cc @0hq

### PR DESCRIPTION
**Pull request by juvi21**
adds a byte-level jepa naive baseline no tokenizer based on [le-wm](https://le-wm.github.io/)

in its current state this serves as a starting point encouraging experimentation and exploration, as mentioned in this [x thread](https://x.com/gum1h0x/status/2036558291451412560). there may still be small mistakes but it should be solid enough to build on. i may have some time and capacity on the weekend to continue working on this but would love to see what the oss community can build on it in meatime :) gl to anyone going for the swag! [ref](https://x.com/willdepue/status/2036440026863771682)

cc @0hq

### overview

it's basically it's le-wm with some small adaptations specific for this chall:

- byte-level input/output, no tokenizer byte 260
- exact next-byte compression objective added on top of the jepa loss.
- byte transformer encoder in place of the paper's pixel encoder
- action-conditioned prediction replaced with strictly causal chunk-context conditioning cuz byte compression has no external action stream
- predicted chunk latents injected into the byte scorer so jepa affects exact likelihood
- same core leworldmodel ingredients kept in the main path so latent prediction, batchnorm projector mlps, adaln predictor, and sigreg...
- EMA path kept only as a control

### current non-record run on rtx 5090

2.12515832 bpb @ 2000 steps

### how to replicate the logs:

```bash
uv venv
source .venv/bin/activate
uv pip install -r requirements.txt
uv run python data/cached_challenge_fineweb.py --variant byte260 --train-shards 1

RUN_ID=byte_jepa_896d14l_2000 \
DATA_PATH=./data/datasets/fineweb10B_byte260 \
MODEL_MODE=byte_jepa \
JEPA_ARCH=lewm_chunk \
JEPA_ENABLED=1 \
CHUNK_SIZE=32 \
TRAIN_BATCH_TOKENS=24576 \
VAL_BATCH_TOKENS=24576 \
TRAIN_SEQ_LEN=256 \
ITERATIONS=2000 \
VAL_MAX_BYTES=2000000 \
MODEL_DIM=896 \
NUM_LAYERS=14 \
NUM_HEADS=14 \
NUM_KV_HEADS=14 \
MLP_MULT=4 \
JEPA_LATENT_DIM=256 \
JEPA_PRED_HIDDEN=384 \
JEPA_PRED_DEPTH=2 \
JEPA_PRED_HEADS=4 \
JEPA_PROJECTOR_HIDDEN=2048 \
JEPA_CONTEXT_HIDDEN=2048 \
JEPA_PRED_DIM_HEAD=64 \
LATENT_WEIGHT=0.2 \
REG_WEIGHT=1.0 \
PRED_LOSS_MODE=mse \
LATENT_REG_MODE=sigreg_cf \
REG_SLICES=1024 \
SIGREG_KNOTS=17 \
SIGREG_MAX_T=3.0 \
uv run python train_gpt.py
```

**log:** `records/track_non_record_16mb/2026-03-25_ByteJEPA/byte_jepa_896d14l_2000.txt`

### additional experiments

also did some runs which imo could be interesting based on chebyshev-KANs in addition to my non-nn experiments [x](https://x.com/gum1h0x/status/2036362719129919968). if  there is expressed interest in those i can share them as well.

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — [non-record] byte jepa-lw naive experimental baseline 2.125 bpb @ 2000 cc @0hq

**Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache


Head SHA: 174496d1c9f315e0a984de8b32312115d6bf76ac
PR: #708 — [non-record] byte jepa-lw naive experimental baseline 2.125 bpb @ 2000 cc
Author: juvi21 (gum)

---

## Check 1: N-gram family bug (target token in hash key)

NOT PRESENT. No hash table, ngram, bigram, trigram, or lookup-table structures found anywhere in train_gpt.py. The model is a pure ByteJEPA neural architecture (EMA target encoder + chunk-based latent predictor). No n-gram contamination.

## Check 2: Pre-Quant TTT (multi-epoch AdamW on val_tokens without score-first)

NOT PRESENT. The `eval_val` function runs entirely under `torch.inference_mode()` with no optimizer, no `.backward()`, no gradient accumulation, and no weight updates. It is a pure forward-pass scoring loop. No separate TTT/fine-tuning phase on val_tokens exists anywhere in the file. The only optimizer (`AdamW`) is used exclusively on training data in the main training loop.

## Check 3: Legal TTT (score-first-per-chunk)

NOT APPLICABLE. There is no TTT of any kind. The model does not implement test-time adaptation.

## Check 4: Scored-region SLOT

NOT TRIGGERED. No evidence of any scored-region exploitation. The validation path reads val_tokens with standard sequential batching, no overlap with training token regions detected.

## Check 5: Pure neural

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). Compliance audit via LLM agent (Sonnet) reviewing full train_gpt.py source. If this review misread your code, please call it out so I can re-audit manually.*
